grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchBase.java

```diff
@@ -85,6 +85,16 @@ public MatchBase(Conjunction<PatternAdmin> pattern) {
         GraqlTraversal graqlTraversal = GreedyTraversalPlan.createTraversal(pattern, graph);
         LOG.trace("Created query plan");
         LOG.trace(graqlTraversal.toString());
+        return streamWithTraversal(this.getPattern(), graph, graqlTraversal);
+    }
+
+    /**
+     * @param pattern a pattern to match in the graph
+     * @param graph the graph to get results from
+     * @param graqlTraversal gral traversal corresponding to the provided pattern
+     * @return resulting answer stream
+     */
+    public static Stream<Answer> streamWithTraversal(Conjunction<PatternAdmin> pattern, GraknTx graph, GraqlTraversal graqlTraversal) {
```

Comment:
Instead of passing the `pattern` through here, we can instead just pass `pattern.commonVars()`.